scripts/pr-previews/cleanup.py

```diff
@@ -39,8 +40,20 @@ def main() -> None:
             logger.info("No changed files detected, so no push made to gh-pages")
             return
 
-        commit_all_and_push("Clean up PR previews")
-        logger.info("Pushed updates to gh-pages branch")
+        # Be extra sure about the branch before we force-push
+        current_branch = run_subprocess(
+            ["git", "branch", "--show-current"]
+        ).stdout.strip()
+        assert (
+            current_branch == "gh-pages"
+        ), f"Did not expect to be on branch '{current_branch}'; exiting"
```

Comment:
This is maybe a bit paranoid, but it's possible a bug introduced to the shared utilities might leave us on a branch we're not expecting.

Reply:
Good idea, done :)